plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -3747,17 +3748,17 @@ private static void validatePartitionColumns(ConnectorTableMetadata tableMetadat
                 .map(ColumnMetadata::getName)
                 .collect(toImmutableList());
 
-        if (!allColumns.containsAll(partitionedBy)) {
-            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Partition columns %s not present in schema", Sets.difference(ImmutableSet.copyOf(partitionedBy), ImmutableSet.copyOf(allColumns))));
-        }
-
         if (allColumns.size() == partitionedBy.size()) {
             throw new TrinoException(INVALID_TABLE_PROPERTY, "Table contains only partition columns");
         }
 
         if (!allColumns.subList(allColumns.size() - partitionedBy.size(), allColumns.size()).equals(partitionedBy)) {
             throw new TrinoException(HIVE_COLUMN_ORDER_MISMATCH, "Partition keys must be the last columns in the table and in the same order as the table properties: " + partitionedBy);
         }
+
+        if (!ImmutableSet.copyOf(allColumns).containsAll(partitionedBy)) {
```

Comment:
This check needs to be above, as before, otherwise the checks after it won't function correctly in the case where this would fail